jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -284,6 +338,51 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
 
         return collapsed_spec
 
+    def get_aperture(self):
+        # Retrieve flux cube and create an array to represent the cone mask
+        flux_cube = self._app._jdaviz_helper._loaded_flux_cube.get_object(cls=Spectrum1D,
+                                                                          statistic=None)
+        # TODO: Replace with code for retrieving display_unit in cubeviz when it is available
+        display_unit = flux_cube.spectral_axis.unit
+
+        # Center is reverse coordinates
+        center = (self.aperture.selected_spatial_region.center.y,
+                  self.aperture.selected_spatial_region.center.x)
+
+        im_shape = (flux_cube.shape[0], flux_cube.shape[1])
+        aper_method = self.aperture_method_selected.lower()
+        if self.wavelength_dependent:
+            # Cone aperture
+            if display_unit.physical_type != 'length':
+                raise ValueError(
+                    f'Spectral axis unit physical type is {display_unit.physical_type}, '
+                    'must be length for cone aperture')
+
+            # TODO: Remove when cone support is extended to other shapes
+            if not hasattr(self.aperture.selected_spatial_region, 'radius'):
+                raise NotImplementedError(
+                    f"{self.aperture.selected_spatial_region.__class__.__name__} is not supported")
+
+            mask_weights = np.zeros_like(flux_cube.flux.value, dtype=np.float32)
+
+            # TODO: Use flux_cube.spectral_axis.to_value(display_unit) when we have unit conversion.
+            radii = ((flux_cube.spectral_axis.value / self.reference_wavelength) *
+                     self.aperture.selected_spatial_region.radius)
+            # Loop through cube and create cone aperture at each wavelength. Then convert that to a
+            # weight array using the selected aperture method, and add it to a weight cube.
+            for index, cone_radius in enumerate(radii):
+                aperture = CircularAperture(center, r=cone_radius)
+                slice_mask = aperture.to_mask(method=aper_method).to_image(im_shape)
+                # Add slice mask to fractional pixel array
+                mask_weights[:, :, index] = slice_mask
+        else:
+            # Cylindrical aperture
+            aperture = regions2aperture(self.aperture.selected_spatial_region)
```

Comment:
@javerbukh , you cannot use this here because this is incompatible with you having to swap X and Y of the center for the cone case. This would result in the aperture being centered wrong, resulting in test failure, which you will be able to see now in the CI because I made your test case off-center just now.

https://github.com/spacetelescope/jdaviz/blob/20de93b7fd7e5689da441be1f5cef2f9cf6bd8db/jdaviz/core/region_translators.py#L170

Reply:
I might have an idea but I can't investigate till after 9:30

Reply:
My thought would be to add a `switch_center` parameter to `regions2aperture` but if you have a cleaner solution please feel free to implement it. If you can't get to it today, let me know and I will work on it.

Reply:
I thought about that too but that feels wrong. I want to see if I can switch it after the fact. Stay tuned.